File: fud-ai/ios/calorietracker/Models/WeightEntry.swift
import Foundation

struct WeightEntry: Identifiable, Codable {
    let id: UUID
    let date: Date
    var weightKg: Double

    init(id: UUID = UUID(), date: Date = .now, weightKg: Double) {
        self.id = id
        self.date = date
        self.weightKg = weightKg
    }

    var weightLbs: Double {
        weightKg * 2.20462
    }
}

File: reference/fud-ai/android/app/src/main/java/com/apoorvdarshan/calorietracker/models/WeightEntry.kt
package com.apoorvdarshan.calorietracker.models

import kotlinx.serialization.Serializable
import java.time.Instant
import java.util.UUID

@Serializable
data class WeightEntry(
    @Serializable(with = UuidSerializer::class)
    val id: UUID = UUID.randomUUID(),
    @Serializable(with = InstantSerializer::class)
    val date: Instant = Instant.now(),
    val weightKg: Double
) {
    val weightLbs: Double get() = weightKg * 2.20462
}
